test/CMakeLists.txt

```diff
@@ -69,47 +57,70 @@ add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/small
+            ${UNIT_TEST_TARGETS}
+    COMMENT "Building unit tests"
+)
+
 add_custom_target(test-unit
     DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/small
             ${UNIT_TEST_TARGETS}
-    COMMAND ${PROJECT_SOURCE_DIR}/test/test-run.py
-        --builddir=${PROJECT_BINARY_DIR}
-        small/
-        unit/)
+    COMMAND ${CMAKE_CTEST_COMMAND} ${TARANTOOL_CTEST_FLAGS}
+            -L unit -L tarantool_small
+    COMMENT "Running tarantool and small unit tests"
+)
 
 add_custom_target(test-unit-force
     DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/small
             ${UNIT_TEST_TARGETS}
-    COMMAND ${PROJECT_SOURCE_DIR}/test/test-run.py
-        --builddir=${PROJECT_BINARY_DIR}
-        --force
-        small/
-        unit/)
+    COMMAND ${CMAKE_CTEST_COMMAND} ${TARANTOOL_CTEST_FLAGS}
+            -L unit -L tarantool_small
+    COMMENT "Running tarantool and small unit tests"
+)
 
 add_custom_target(test-func
-    COMMAND ${PROJECT_SOURCE_DIR}/test/test-run.py
-        --builddir=${PROJECT_BINARY_DIR}
-        --exclude small/
-        --exclude unit/)
+    COMMAND ${CMAKE_CTEST_COMMAND} ${TARANTOOL_CTEST_FLAGS}
+            --stop-on-failure -LE unit -LE long -LE tarantool_small
+    COMMENT "Running functional regression tests (wo small, unit and long)"
+)
 
 add_custom_target(test-func-force
-    COMMAND ${PROJECT_SOURCE_DIR}/test/test-run.py
-        --builddir=${PROJECT_BINARY_DIR}
-        --exclude small/
-        --exclude unit/
-        --force)
+    COMMAND ${CMAKE_CTEST_COMMAND} ${TARANTOOL_CTEST_FLAGS}
+            -LE unit -LE long -LE tarantool_small
+    COMMENT "Running functional regression tests (wo small, unit and long)"
+)
 
-add_custom_target(test
-    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/small
-            LuaJIT-test
-            ${UNIT_TEST_TARGETS}
-    COMMAND ${PROJECT_SOURCE_DIR}/test/test-run.py
-        --builddir=${PROJECT_BINARY_DIR})
+add_custom_target(test-all
+    DEPENDS ${UNIT_TEST_TARGETS}
+    COMMAND ${CMAKE_CTEST_COMMAND} ${TARANTOOL_CTEST_FLAGS}
+            --stop-on-failure -LE long
+    COMMENT "Running all regression tests"
+)
 
 add_custom_target(test-force
     DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/small
-            LuaJIT-test
             ${UNIT_TEST_TARGETS}
-    COMMAND ${PROJECT_SOURCE_DIR}/test/test-run.py
-        --builddir=${PROJECT_BINARY_DIR}
-        --force)
+    COMMAND ${CMAKE_CTEST_COMMAND} ${TARANTOOL_CTEST_FLAGS}
+    COMMENT "Running regression tests"
+)
+
+message(STATUS "Add integration tests")
+file(GLOB_RECURSE tests ${PROJECT_SOURCE_DIR}/tests *.test.lua *.test.py *_test.lua)
+foreach(test_path ${tests})
+  get_filename_component(test_name ${test_path} NAME)
+  get_filename_component(basedir ${test_path} DIRECTORY)
+  get_filename_component(suite ${basedir} NAME)
+  set(test_title "test/${suite}/${test_name}")
+  add_test(NAME ${test_title}
+           COMMAND ${PROJECT_SOURCE_DIR}/test/test-run.py
+                   --builddir=${PROJECT_BINARY_DIR}
+                   --debug
```

Comment:
By default, `ctest` suppress test stdout/stderr, and if test is failed `ctest` prints all stdout/stderr returned by the test. I want to see a maximum of information from test-run in case of test fail.
So the answer is no.

Reply:
I just afraid that this is too much unrelated debug info. Feel free to ignore.